LibraBFT/Impl/Consensus/RoundManager.agda

```diff
@@ -32,48 +33,76 @@ open RWST-do
 
 ------------------------------------------------------------------------------
 
+processCommitM : LedgerInfoWithSignatures → LBFT (List ExecutedBlock)
+processCommitM finalityProof = pure []
+
+-- IMPL-TODO: implement this
 processNewRoundEventM : Instant → NewRoundEvent → LBFT Unit
 processNewRoundEventM now nre = pure unit
 
 ------------------------------------------------------------------------------
 
-syncUpM               : Instant → SyncInfo → Author                   → LBFT (ErrLog ⊎ Unit)
-ensureRoundAndSyncUpM : Instant → Round    → SyncInfo → Author → Bool → LBFT (ErrLog ⊎ Bool)
+ensureRoundAndSyncUpM : Instant → Round    → SyncInfo → Author → Bool → LBFT (FakeErr ⊎ Bool)
 processProposalM      : Block                                         → LBFT Unit
-executeAndVoteM       : Block                                         → LBFT (ErrLog ⊎ VoteWithMeta)
+executeAndVoteM       : Block                                         → LBFT (FakeErr ⊎ Vote)
 
 -- external entry point
 -- TODO-2: The sync info that the peer requests if it discovers that its round
 -- state is behind the sender's should be sent as an additional argument, for now.
-processProposalMsgM : Instant → {- Author → -} ProposalMsg → LBFT Unit
-processProposalMsgM now {- from -} pm =
-  case pm ^∙ pmProposer of λ where
-    nothing        → pure unit -- log: info: proposal with no author
-    (just pAuthor) →
-      ensureRoundAndSyncUpM now (pm ^∙ pmProposal ∙ bRound) (pm ^∙ pmSyncInfo)
-                            pAuthor true >>= λ where
-      (inj₁ _)     → pure unit -- log: error: <propagate error>
-      (inj₂ true)  → processProposalM (pm ^∙ pmProposal)
-      (inj₂ false) → do
-        currentRound ← use (lRoundState ∙ rsCurrentRound)
-        pure unit -- log: info: dropping proposal for old round
+module processProposalMsgM (now : Instant) (pm : ProposalMsg) where
+  step₀ : LBFT Unit
+  step₁ : Author → LBFT Unit
+  step₂ : FakeErr ⊎ Bool → LBFT Unit
+
+  step₀ =
+    case pm ^∙ pmProposer of λ where
+      nothing → logInfo -- log: info: proposal with no author
+      (just pAuthor) → step₁ pAuthor
+  step₁ pAuthor =
+        ensureRoundAndSyncUpM now (pm ^∙ pmProposal ∙ bRound) (pm ^∙ pmSyncInfo)
+                              pAuthor true >>= step₂
+  step₂ r =
+        -- IMPL-DIFF: We use `ifM` to test whether the round of the proposal is
+        -- current, to take advantage of the obligations `RWST-weakestPre` generates.
+        case r of λ where
+        (inj₁ _) → logErr -- log: error: <propagate error>
```

Comment:
I propose that, rather than re-defining disjoint union as `Either`, we make an alias and define `Left` and `Right` as patterns. https://agda.readthedocs.io/en/v2.5.2/language/pattern-synonyms.html

This way, we maintain re-usability and readability.

Reply:
OK, I have a commit in which I did this the "wrong" way, but also fixed all the relevant `inj₁`s to `Left`, etc.  Hopefully the latter grunt work can remain.  I'll experiment with doing this the "right" way in that context and see if I can make it work.  Thanks.